raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansGrouper.java

```diff
@@ -1,26 +1,120 @@
+import org.apache.kafka.streams.state.internals.ValueAndTimestampSerde;
+import org.hypertrace.core.datamodel.RawSpan;
+import org.hypertrace.core.datamodel.RawSpans;
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.kafkastreams.framework.KafkaStreamsApp;
 import org.hypertrace.core.serviceframework.config.ConfigClient;
+import org.hypertrace.core.serviceframework.config.ConfigUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class RawSpansGrouper extends PlatformBackgroundService {
-  private static Logger LOGGER = LoggerFactory.getLogger(RawSpansGrouper.class);
+public class RawSpansGrouper extends KafkaStreamsApp {
+
+  private static final Logger logger = LoggerFactory
+      .getLogger(RawSpansGrouper.class);
 
   public RawSpansGrouper(ConfigClient configClient) {
     super(configClient);
   }
 
+  public StreamsBuilder buildTopology(Map<String, Object> properties, StreamsBuilder streamsBuilder,
+      Map<String, KStream<?, ?>> inputStreams) {
+    Config jobConfig = getJobConfig(properties);
+    String inputTopic = jobConfig.getString(INPUT_TOPIC_CONFIG_KEY);
+    String outputTopic = jobConfig.getString(OUTPUT_TOPIC_CONFIG_KEY);
+
+    KStream<String, RawSpan> inputStream = (KStream<String, RawSpan>) inputStreams.get(inputTopic);
+    if (inputStream == null) {
+      inputStream = streamsBuilder
+          // read the input topic
+          .stream(inputTopic, Consumed.with(Serdes.String(), null));
+      inputStreams.put(inputTopic, inputStream);
+    }
+
+    // Retrieve the default value serde defined in config and use it
+    Serde valueSerde = defaultValueSerde(properties);
+    StoreBuilder<KeyValueStore<String, ValueAndTimestamp<RawSpans>>> traceStoreBuilder = Stores
+        .keyValueStoreBuilder(Stores.persistentKeyValueStore(INFLIGHT_TRACE_STORE),
+            Serdes.String(),
+            new ValueAndTimestampSerde<>(valueSerde));
+    StoreBuilder<KeyValueStore<String, Long>> traceEmitTriggerStoreBuilder = Stores
+        .keyValueStoreBuilder(Stores.persistentKeyValueStore(TRACE_EMIT_TRIGGER_STORE),
+            Serdes.String(), Serdes.Long());
+
+    streamsBuilder.addStateStore(traceStoreBuilder);
+    streamsBuilder.addStateStore(traceEmitTriggerStoreBuilder);
+
+    Produced<String, StructuredTrace> outputTopicProducer = Produced.with(Serdes.String(), null);
+    outputTopicProducer = outputTopicProducer.withName(OUTPUT_TOPIC_PRODUCER);
+
+    inputStream
+        .transform(RawSpansGroupingTransformer::new,
+            Named.as(RawSpansGroupingTransformer.class.getSimpleName()), INFLIGHT_TRACE_STORE,
+            TRACE_EMIT_TRIGGER_STORE)
+        .to(outputTopic, outputTopicProducer);
+
+    return streamsBuilder;
+  }
+
   @Override
-  protected PlatformBackgroundJob createBackgroundJob(Config config) {
-    return new RawSpanToStructuredTraceGroupingJob(config);
+  public Map<String, Object> getStreamsConfig(Config config) {
```

Comment:
We don't need this method as same impl is moved to base class.

Reply:
fixed